models/model_upload/multimodal_models/sglang-llama-3_2-11b-instruct/1/model.py

```diff
@@ -0,0 +1,240 @@
+  inference_params = {}
+  if request.model.model_version.id != "":
+    output_info = request.model.model_version.output_info
+    output_info = json_format.MessageToDict(output_info, preserving_proto_field_name=True)
+
+    if "params" in output_info:
+      inference_params = output_info["params"]
+  return inference_params
+
+
+# All CLI arguments for the sglang server are defined here.
+# https://github.com/sgl-project/sglang/blob/v0.3.6/python/sglang/srt/server_args.py#L210
+class SGLangServerManager:
+
+  def __init__(self,
+               mem_fraction_static,
+               tensor_parallel_size,
+               dtype="auto",
+               context_length=None,
+               quantization=None,
+               chat_template=None,
+               port=8761,
+               host="localhost"):
+    self.mem_fraction_static = mem_fraction_static
+    self.tensor_parallel_size = tensor_parallel_size
+    self.context_length = context_length
+    self.quantization = quantization
+    self.dtype = dtype
+    self.chat_template = chat_template
+    self.host = host
+    self.port = port
+
+    self.command = None
+    self.process = None
+
+  def start_server(self, python_executable, checkpoints):
+    """Start the sglang server."""
+    self.command = f"{python_executable} -m sglang.launch_server --model-path {checkpoints}"
+    if self.dtype:
+      self.command += f" --dtype {self.dtype}"
+    if self.quantization:
+      self.command += f" --quantization {self.quantization}"
+    if self.tensor_parallel_size:
+      self.command += f" --tensor-parallel-size {self.tensor_parallel_size}"
+    if self.mem_fraction_static:
+      self.command += f" --mem-fraction-static {self.mem_fraction_static}"
+    if self.context_length:
+      self.command += f" --context-length {self.context_length}"
+    if self.chat_template:
+      self.command += f" --chat-template {self.chat_template}"
+    if self.host:
+      self.command += f" --host {self.host}"
+    if self.port:
+      self.command += f" --port {self.port}"
+
+    try:
+      self.process = execute_shell_command(self.command)
+      wait_for_server(f'http://localhost:{self.port}')
+    except Exception as e:
+      if self.process:
+        logger.error("Terminating the sglang server process.")
+        terminate_process(self.process)
+      raise RuntimeError("Failed to start sglang server: " + str(e))
+
+
+def stream_completion(model, client, input_data, inference_params):
+  """Stream iteratively generates completions for the input data."""
+
+  temperature = inference_params.get("temperature", 0.7)
+  max_tokens = inference_params.get("max_tokens", 512)
+  top_p = inference_params.get("top_p", 1.0)
+
+  messages = construct_messages(input_data)
+  kwargs = dict(
+      model=model,
+      messages=messages,
+      temperature=temperature,
+      max_tokens=max_tokens,
+      top_p=top_p,
+      extra_body={"stop_token_ids": [151645, 151643]},
```

Comment:
This one is mysterious to me. If I change the model, do I need to change these? You may want to leave a comment here.